cloudwatch_logger/include/cloudwatch_logger/log_node.h

```diff
@@ -49,9 +51,11 @@ class LogNode
    * @param log_stream log stream name
    * @param config aws client configuration object
    * @param sdk_options aws sdk options
+   * @param log_manager_factory optional log manager factory
    */
   void Initialize(const std::string & log_group, const std::string & log_stream,
-                  Aws::Client::ClientConfiguration & config, Aws::SDKOptions & sdk_options);
+                  const Aws::Client::ClientConfiguration & config, Aws::SDKOptions & sdk_options, 
+                  const std::shared_ptr<LogManagerFactory> & log_manager_factory = std::make_shared<LogManagerFactory>()); 
```

Comment:
I don't quite understand what it means to pass a shared_ptr by const reference here.

Reply:
I don't think it's safe to pass this by reference, because what if the last strong pointer to the `LogManagerFactory` that's passed in is decremented and deallocated from a separate thread?

Reply:
I've changed it to pass by value

Reply:
Thanks.

Just wanted to clarify that passing the shared_ptr by reference wasn't any less than safe than passing all those other parameters by reference, but it defeats the purpose of shared_ptr.